### PR TITLE
Don't do partition filter check on analyze call

### PR DESCRIPTION
**Pull request by JamesRTaylor**

### CODE REVIEW

**Comment by JamesRTaylor** [comment]
Do you think `ANALYZE <table>` should be allowed with the partition filter check enabled (i.e. without specifying specific partitions)? That's currently not allowed.

**Comment by martint** [comment]
Probably not. That’s effectively a full table scan and has the same cost of running a query without a partition filter

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 1]
```diff
@@ -434,6 +434,31 @@ public void testPartitionPredicateDisallowed() throws Exception
...
+                        .build())
+                .setCatalogSessionProperty("hive", "query_partition_filter_required", "true")
+                .build();
+
+        assertUpdate(
+                session,
+                "create table partition_test14(\n"
+                        + "id integer,\n"
+                        + "a varchar,\n"
+                        + "b varchar,\n"
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: replace `\n` with ` ` for readability

(you can also put all columns on one line)

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 2]
```diff
@@ -434,6 +434,31 @@ public void testPartitionPredicateDisallowed() throws Exception
...
+                        + "ds varchar)"
+                        + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
+        assertUpdate(session, "insert into partition_test14(id,a,ds) values(1, 'a','a')", 1);
+        String query = "analyze partition_test14 WITH (PARTITIONS=ARRAY[ARRAY['a']])";
+        computeActual(session, query);
+        computeActual(session, "explain " + query);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]:
```suggestion
        computeActual(session, "EXPLAIN " + query);
```

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 3]
```diff
@@ -434,6 +434,31 @@ public void testPartitionPredicateDisallowed() throws Exception
...
+        assertUpdate(
+                session,
+                "create table partition_test14(\n"
+                        + "id integer,\n"
+                        + "a varchar,\n"
+                        + "b varchar,\n"
+                        + "ds varchar)"
+                        + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
+        assertUpdate(session, "insert into partition_test14(id,a,ds) values(1, 'a','a')", 1);
+        String query = "analyze partition_test14 WITH (PARTITIONS=ARRAY[ARRAY['a']])";
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]:
```suggestion
        String query = "ANALYZE partition_test14 WITH (PARTITIONS=ARRAY[ARRAY['a']])";
```

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 4]
```diff
@@ -434,6 +434,31 @@ public void testPartitionPredicateDisallowed() throws Exception
...
+
+        assertUpdate(
+                session,
+                "create table partition_test14(\n"
+                        + "id integer,\n"
+                        + "a varchar,\n"
+                        + "b varchar,\n"
+                        + "ds varchar)"
+                        + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
+        assertUpdate(session, "insert into partition_test14(id,a,ds) values(1, 'a','a')", 1);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]:
```suggestion
        assertUpdate(session, "INSERT INTO partition_test14(id,a,ds) VALUES (1, 'a','a')", 1);
```

**Comment by JamesRTaylor** [comment]
Thanks for the review, @findepi. I've incorporated your feedback and pushed a new commit.

**Pull request merged**